### PR TITLE
Feat/text box

### PR DESCRIPTION
**Pull request by pedrohgsilveira**
# PR Description

#### Proposed changes

Adds the Text Box and implements it

#### Kind of Change
What kind of change does your code introduce? Put an ```x``` in the box that applies.
**NOTE: Please do not create MRs with different purposes. Each MR must have a single purpose: Hotfix, New Feature, or Structural Change.**

- [ ] Hotfix.
- [x] New Feature.
- [x] Structural Change.

#### Checklist

Put an ```x``` in the boxes that apply.

### This MR follows the base conventions of the project:
- [x] Does not add commented code.
- [x] Does not add code with ```print```.
- [x] Documentation have been added / updated as needed.
- [x] New Strings are located in ```*.strings``` file.
- [x] IBOutlets and delegates should be weak to avoid retain cycles.

### CODE REVIEW

File: MacroPepelelipa/MacroPepelelipa/Controller/Notebook/Notes/NotesViewController.swift [hunk 1]
```diff
@@ -160,4 +143,176 @@ internal class NotesViewController: UIViewController, TextEditingDelegateObserve
             textField.trailingAnchor.constraint(equalTo: self.view.safeAreaLayoutGuide.trailingAnchor, constant: -20)
         ])
     }
+    
+    func textEditingDidBegin() {
+        DispatchQueue.main.async {
+            self.textBoxes.forEach { (textBox) in
+                textBox.state = .idle
+                textBox.markupTextView.isUserInteractionEnabled = false
+                textBox.backgroundColor = .red
```
Comment (PedroFarina) [inline MacroPepelelipa/MacroPepelelipa/Controller/Notebook/Notes/NotesViewController.swift:152]: ?????

File: MacroPepelelipa/MacroPepelelipa/View/Markup/MarkupToolBar.swift [hunk 2]
```diff
@@ -26,7 +26,7 @@ enum HeaderStyle {
 internal class MarkupToolBar: UIToolbar {
     
     private weak var textView: MarkupTextView?
-    private weak var viewController: UIViewController?
+    private weak var viewController: NotesViewController?
```
Comment (PedroFarina) [inline MacroPepelelipa/MacroPepelelipa/View/Markup/MarkupToolBar.swift:29]: exposing a whole class to use only one method?

**Review by liakassardjian — COMMENTED** [review]
Ou é New Feature, ou é Structural Change, um dos dois

**Review by PedroFarina — CHANGES_REQUESTED** [review]
Change exposure of NotesViewController

**Review by PedroFarina — APPROVED** [review]

**Pull request merged**